engine/rpcserver.go

```diff
@@ -1213,3 +1221,333 @@ func (s *RPCServer) GetAuditEvent(ctx context.Context, r *gctrpc.GetAuditEventRe
 
 	return &resp, nil
 }
+
+// GCTScriptStatus returns a slice of current running scripts that includes next run time and uuid
+func (s *RPCServer) GCTScriptStatus(ctx context.Context, r *gctrpc.GCTScriptStatusRequest) (*gctrpc.GCTScriptStatusResponse, error) {
+	if !gctscript.GCTScriptConfig.Enabled {
+		return &gctrpc.GCTScriptStatusResponse{Status: gctscript.ErrScriptingDisabled.Error()}, nil
+	}
+
+	totalVM := len(gctscript.AllVMs)
+
+	if totalVM < 1 {
+		return &gctrpc.GCTScriptStatusResponse{Status: "no scripts running"}, nil
+	}
+
+	resp := &gctrpc.GCTScriptStatusResponse{
+		Status: fmt.Sprintf("%v of %v virtual machines running", totalVM, gctscript.GCTScriptConfig.MaxVirtualMachines),
+	}
+
+	for x := range gctscript.AllVMs {
+		resp.Scripts = append(resp.Scripts, &gctrpc.GCTScript{
+			UUID:    gctscript.AllVMs[x].ID.String(),
+			Name:    gctscript.AllVMs[x].ShortName(),
+			NextRun: gctscript.AllVMs[x].NextRun.String(),
+		})
+	}
+
+	return resp, nil
+}
+
+// GCTScriptQuery queries a running script and returns script running information
+func (s *RPCServer) GCTScriptQuery(ctx context.Context, r *gctrpc.GCTScriptQueryRequest) (*gctrpc.GCTScriptQueryResponse, error) {
+	if !gctscript.GCTScriptConfig.Enabled {
+		return &gctrpc.GCTScriptQueryResponse{Status: gctscript.ErrScriptingDisabled.Error()}, nil
+	}
+
+	UUID, err := uuid.FromString(r.Script.UUID)
+	if err != nil {
+		return &gctrpc.GCTScriptQueryResponse{Status: "error", Data: err.Error()}, nil
+	}
+
+	if v, f := gctscript.AllVMs[UUID]; f {
+		resp := &gctrpc.GCTScriptQueryResponse{
+			Status: "ok",
+			Script: &gctrpc.GCTScript{
+				Name:    v.ShortName(),
+				UUID:    v.ID.String(),
+				Path:    v.Path,
+				NextRun: v.NextRun.String(),
+			},
+		}
+		data, err := v.Read()
+		if err != nil {
+			return nil, err
+		}
+		resp.Data = string(data)
+		return resp, nil
+	}
+	return &gctrpc.GCTScriptQueryResponse{Status: "not found", Data: "UUID not found"}, nil
+}
+
+// GCTScriptExecute execute a script
+func (s *RPCServer) GCTScriptExecute(ctx context.Context, r *gctrpc.GCTScriptExecuteRequest) (*gctrpc.GCTScriptGenericResponse, error) {
+	if !gctscript.GCTScriptConfig.Enabled {
+		return &gctrpc.GCTScriptGenericResponse{Status: gctscript.ErrScriptingDisabled.Error()}, nil
+	}
+
+	if r.Script.Path == "" {
+		r.Script.Path = gctscript.ScriptPath
+	}
+
+	gctVM := gctscript.New()
+	if gctVM == nil {
+		return &gctrpc.GCTScriptGenericResponse{Status: "error", Data: "unable to create VM instance"}, nil
+	}
+
+	script := filepath.Join(r.Script.Path, r.Script.Name)
+	err := gctVM.Load(script)
+	if err != nil {
+		return &gctrpc.GCTScriptGenericResponse{
+			Status: "error",
```

Comment:
Yep I agree
Have been cleaning up all the responses today in between dealing with my roof :D

at the moment the are:
```
	// MsgOK message to display when status is "OK"
	MsgStatusOK string = "ok"
	// MsgStatusSuccess message to display when status is successful
	MsgStatusSuccess string = "success"
	// MsgStatusError message to display when failure occurs
	MsgStatusError string = "error"
```

I am not sure if the Msg is needed but because they sit in engine and are exported I didn't want to cause confusion 

Maybe some generic responses package that imports from nothing and is just purely exported commonly used messages might be a good idea thoughts @gloriousCode ?

Reply:
I'm happy with a common string package if its kept small. Otherwise to say "error" we import 32767 strings.
I'm happy with 'msg' prefix being there since we normally have an 'err' prefix for err message consts, but I'm also happy if its not there since it doesn't provide any extra insight.